File: src/main/java/ru/trukhmanov/twochairsbackend/security/JwtService.java
package ru.trukhmanov.twochairsbackend.security;

import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.security.Keys;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import javax.crypto.SecretKey;
import java.nio.charset.StandardCharsets;
import java.time.Instant;
import java.util.Date;

@Service
public class JwtService {

    private final SecretKey key;
    private final String issuer;
    private final long ttlSeconds;

    public JwtService(
            @Value("${spring.security.jwt.secret}") String secret,
            @Value("${spring.security.jwt.issuer}") String issuer,
            @Value("${spring.security.jwt.accessTokenTtlSeconds}") long ttlSeconds
    ) {
        this.key = Keys.hmacShaKeyFor(secret.getBytes(StandardCharsets.UTF_8));
        this.issuer = issuer;
        this.ttlSeconds = ttlSeconds;
    }

    public String issueToken(long userId, String phoneNumber) {
        Instant now = Instant.now();
        Instant exp = now.plusSeconds(ttlSeconds);

        return Jwts.builder()
                .issuer(issuer)
                .subject(Long.toString(userId))
                .claim("phone", phoneNumber)
                .issuedAt(Date.from(now))
                .expiration(Date.from(exp))
                .signWith(key)
                .compact();
    }
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/security/JwtService.kt
package ru.trukhmanov.twochairsbackend.security

import io.jsonwebtoken.Jwts
import io.jsonwebtoken.security.Keys
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import java.nio.charset.StandardCharsets
import java.time.Instant
import java.util.Date
import javax.crypto.SecretKey

@Service
class JwtService(
    @Value("\${spring.security.jwt.secret}") secret: String,
    @param:Value("\${spring.security.jwt.issuer}") private val issuer: String,
    @param:Value("\${spring.security.jwt.accessTokenTtlSeconds}") private val ttlSeconds: Long
) {

    private val key: SecretKey = Keys.hmacShaKeyFor(secret.toByteArray(StandardCharsets.UTF_8))

    fun issueToken(userId: Long, phoneNumber: String): String {
        val now = Instant.now()
        val exp = now.plusSeconds(ttlSeconds)

        return Jwts.builder()
            .issuer(issuer)
            .subject(userId.toString())
            .claim("phone", phoneNumber)
            .issuedAt(Date.from(now))
            .expiration(Date.from(exp))
            .signWith(key)
            .compact()
    }
}
